addon-test-support/RouteInfoMock.ts

```diff
@@ -0,0 +1,75 @@
+interface RouteInfo {
+  readonly name: string;
+  readonly parent: RouteInfo | null;
+  readonly child: RouteInfo | null;
+  readonly localName: string;
+  readonly params: { [name: string]: unknown } | undefined;
+  readonly paramNames: string[];
+  readonly queryParams: { [name: string]: string };
+  readonly metadata: unknown;
+  find(
+    predicate: (this: any, routeInfo: RouteInfo, i: number) => boolean,
+    thisArg?: any
+  ): RouteInfo | undefined;
+}
+
+export default class RouteInfoMock implements RouteInfo {
+  name: string;
+  parent: RouteInfo | null = null;
+  child: RouteInfo | null = null;
+  localName: string;
+  params: { [name: string]: unknown } = {};
```

Comment:
Making the above change to use the type from Ember highlights that this isn't quite right and should instead be:

```suggestion
  params: Record<string, string> = {};
```